extensions/flags/extend.php

```diff
@@ -51,27 +41,26 @@
     (new Extend\Model(Post::class))
         ->hasMany('flags', Flag::class, 'post_id'),
 
-    (new Extend\ApiSerializer(PostSerializer::class))
-        ->hasMany('flags', FlagSerializer::class)
-        ->attribute('canFlag', AddCanFlagAttribute::class),
+    new Extend\ApiResource(FlagResource::class),
 
-    (new Extend\ApiSerializer(CurrentUserSerializer::class))
-        ->attribute('newFlagCount', AddNewFlagCountAttribute::class),
+    (new Extend\ApiResource(Resource\PostResource::class))
+        ->fields(PostResourceFields::class),
 
-    (new Extend\ApiSerializer(ForumSerializer::class))
-        ->attributes(AddFlagsApiAttributes::class),
+    (new Extend\ApiResource(Resource\UserResource::class))
+        ->fields(UserResourceFields::class),
 
-    (new Extend\ApiController(ShowDiscussionController::class))
-        ->addInclude(['posts.flags', 'posts.flags.user']),
+    (new Extend\ApiResource(Resource\ForumResource::class))
+        ->fields(ForumResourceFields::class),
 
-    (new Extend\ApiController(ListPostsController::class))
-        ->addInclude(['flags', 'flags.user']),
+    (new Extend\ApiResource(Resource\DiscussionResource::class))
+        ->endpoint(Endpoint\Show::class, function (Endpoint\Show $endpoint) {
```

Comment:
docs actually recommend avoiding default includes whenever possible. Otherwise endpoints become unnecessarily bloated which easily harms performance. I have noticed for example some extensions adding default includes to endpoint A in order to display a relation in frontend page X, only for frontend page B to now unnecessarily have an api call that includes a relation it does not need (which often leads to more issues when you consider lack of eager loading).

With multiple extensions, it all builds up to a pretty bad effect.

Reply:
Yeah this makes sense. I meant more like "default includes are almost always bad, here's why, here's what might qualify as an exception but seriously please don't"